python/packages/ai/teams/ai/ai_options.py

```diff
@@ -15,14 +17,26 @@
 
 @dataclass
 class AIOptions(Generic[StateT]):
+    planner: Planner
+    """
+    The planner to use for generating plans. For example,
+    you could set this as an instance of `OpenAIPlanner` or
+    `AzureOpenAIPlanner`.
+    """
+
+    prompt_manager: PromptManager[StateT] = PromptManager[StateT]()
```

Comment:
Per steve's design, prompt manage is now part of planner and not exposed to user. Can remove this option.

Reply:
because the planner isn't using generics like the rest of the codebase it is throwing type errors so I can't use it for this until it is refactored to use a generic turn state 
![image](https://github.com/microsoft/teams-ai/assets/44811138/179591eb-2802-4e06-8458-10018fd77c61)

Reply:
I see, you're declaring a single StateT instead of using 3 (ConversationT, UserT, TempT), seems mypy cannot handle this.
I'm good to go with current implementation. But I will create new PR to update planner as well as change the implementation to use planner. Let me know if you have any concerns to this.

Reply:
I think what you said below about using the three generics makes sense and if you are in aligment we can merge this PR as is for now and Ill open a new one to change all our `StateT` generics to use your method (3 generics) instead.